src/MagiaClient.cpp

```diff
@@ -211,6 +214,29 @@ const std::string* urlConfigResource(void* a1, UrlConfigResourceType type) {
     return urlConfigResourceHooked(a1, type);
 }
 
+// Redirect to api proxy
```

Comment:
I don't think we can just start redirecting any API requests from users who don't explicitly participate in Puella Historia.

Reply:
Yeah that's why I didn't publish an APK. Maybe some prompt about whether to join this collective effort can be added, so that potentially we (or "your project" if speaking more frankly, I'm egoistic enough to feel uncomfortable to solely rely on you or follow your project 😈 I still want to do it in my own way) can recruit more people (which is positive and I'm for it as well ☺️)

Reply:
The game shuts down in a month, is this intended to use a private server?

Reply:
Yes and no.

Yes because there's an ongoing collective effort to walk through all reachable game contents so that we can preserve the data. No because, um, personally I am not sure whether to make a private server or simply a more feature-enabled offline archival app. (Like, its data should be at least easily exportable for backup and restore, and, official announcement said that stories and quests will not available in archive app, therefore we have to at least enable this with our own effort)

Yes because this pr might help but no because...

There's already proxified APK made by simple hex patch to change endpoint domain name therefore the non-resource traffic can be recorded when going through the proxy server.

However I made this out of some mania actually because there's nobody asking for this. The benefits of this pr is questionable because @LiviaMedeiros already figured out a much simpler way to deal with country code issue and transfer - to decline `api/user/create` for the first time and show the device id to the user, so that the user can complete `api/user/create` externally in a browser to fit his/her own country code.

🙁 Livia doesn't like my approach of js injection because it doesn't seem reliable. I agree that injection might introduce some unknown risk, and it must be taken into consideration that we must avoid provoking f4samurai, we must avoid making unusual behavior or tripping something.

Reply:
Initially I proposed a pure server side handling approach to deal with country code issues: the proxy server may try different country code in turn for the user, therefore this obstacle is conquered seamlessly even without user awareness. However this requires IP addresses of many countries/regions which also have to be not on f4samurai's blacklist. Let alone the effort to be invested.

Reply:
All involved players share one common IP address, the proxy's IP address. Therefore if injected js triggers some error or anormaly it might provoke f4samurai which may eventually lead to account ban. And there's new game (madoka exedra) which can transfer something from magireco. Let alone the risk of losing last chance to preserve more personal account data...

Reply:
However, however I still argued that maybe, maybe...  oh I don't know whether the external browser-based transfer is more risky than js code injection I argued that purely out of egoism 🙁

Reply:
Oh there's way to avoid both js injection and browser-based hack - transplant the device id:

https://gist.github.com/segfault-bilibili/c7090af5958d768ca2caae4e27a11cfa

However this transplanting requires root...

Reply:
> is this intended to use a private server?

Back to this basic question, VirusLord said the English translation will go on after official server closure. Therefore... Is this pr going to repurpose the currently existing MagiaTranslate client into a client of private server? I don't have clear idea for now to be honest. But I think the ability to change endpoint in a runtime hook might help.

Reply:
Personally I prefer to an offline archive app with more complete functionalities. I'm even thinking of nullifying the game's own asset downloading/updating function - instead we make updatable zip packages which naturally survives any server closure, official or private. And, zip files may be placed outside the app's private data directory, therefore it will be naturally easy to backup/import/export - with some virtual file system there would be no import/export at all 😋

However this requires both hooking of both network layer and local fs layer to redirect (like) `connect()` and `fopen()` syscalls, and, a virtual file system to extract files from zip on-the-fly.

Reply:
I think zip packages have other minor advantages like, being one big chunk of data, it downloads/copies in a sequential manner which is generally much faster, and, this will avoid storage space waste when using exFAT with big cluster size. (oh then what the actual usage scenario? Maybe a OTG USB thumb flash drive or a portable disk?)

Reply:
The file name of zip can also contain metadata like language (original JP or community translated EN/CN etc) or version. (However when there's update it seems to have disadvantages, like, superseded files still take storage space? And the update won't be live/instantly effective, it must take effect after downloading new updated zip package)

Reply:
https://discord.com/channels/750791982065909872/1247281094433439826/1257878874239537194